samples/hello-cube.html

```diff
@@ -0,0 +1,336 @@
+
+struct Uniform
+{
+    device float4x4* modelViewProjectionMatrix [[id(0)]];
+};
+
+vertex FragmentData vertex_main(Vertex vertexIn [[stage_in]],
+                          const device Uniform& uniforms [[buffer(1)]])
+{
+    FragmentData output;
+    output.position = uniforms.modelViewProjectionMatrix[0] * vertexIn.position;
+    output.color = vertexIn.color;
+
+    return output;
+}
+
+fragment float4 fragment_main(FragmentData data [[stage_in]])
+{
+    return float4(data.color);
+}
+`
+
+// A * indicates that this symbol has yet to be implemented in WebKit.
+
+let device, context, verticesBuffer, mappedTransformArray, uniformsBindGroup, pipeline, renderPassDescriptor, queue;
+let projectionMatrix = mat4.create();
+
+const vertexSize = 4 * 8;
+const verticesSize = vertexSize * 36;
+const colorOffset = 4 * 4;
+function populateVerticesArray(array) {
+    const vertexData = [
+        // float4 position, float4 color
+        1, -1, 1, 1, 1, 0, 1, 1,
+        -1, -1, 1, 1, 0, 0, 1, 1,
+        -1, -1, -1, 1, 0, 0, 0, 1,
+        1, -1, -1, 1, 1, 0, 0, 1,
+        1, -1, 1, 1, 1, 0, 1, 1,
+        -1, -1, -1, 1, 0, 0, 0, 1,
+
+        1, 1, 1, 1, 1, 1, 1, 1,
+        1, -1, 1, 1, 1, 0, 1, 1,
+        1, -1, -1, 1, 1, 0, 0, 1,
+        1, 1, -1, 1, 1, 1, 0, 1,
+        1, 1, 1, 1, 1, 1, 1, 1,
+        1, -1, -1, 1, 1, 0, 0, 1,
+
+        -1, 1, 1, 1, 0, 1, 1, 1,
+        1, 1, 1, 1, 1, 1, 1, 1,
+        1, 1, -1, 1, 1, 1, 0, 1,
+        -1, 1, -1, 1, 0, 1, 0, 1,
+        -1, 1, 1, 1, 0, 1, 1, 1,
+        1, 1, -1, 1, 1, 1, 0, 1,
+
+        -1, -1, 1, 1, 0, 0, 1, 1,
+        -1, 1, 1, 1, 0, 1, 1, 1,
+        -1, 1, -1, 1, 0, 1, 0, 1,
+        -1, -1, -1, 1, 0, 0, 0, 1,
+        -1, -1, 1, 1, 0, 0, 1, 1,
+        -1, 1, -1, 1, 0, 1, 0, 1,
+
+        1, 1, 1, 1, 1, 1, 1, 1,
+        -1, 1, 1, 1, 0, 1, 1, 1,
+        -1, -1, 1, 1, 0, 0, 1, 1,
+        -1, -1, 1, 1, 0, 0, 1, 1,
+        1, -1, 1, 1, 1, 0, 1, 1,
+        1, 1, 1, 1, 1, 1, 1, 1,
+
+        1, -1, -1, 1, 1, 0, 0, 1,
+        -1, -1, -1, 1, 0, 0, 0, 1,
+        -1, 1, -1, 1, 0, 1, 0, 1,
+        1, 1, -1, 1, 1, 1, 0, 1,
+        1, -1, -1, 1, 1, 0, 0, 1,
+        -1, 1, -1, 1, 0, 1, 0, 1,
+    ];
+
+    for (let i = 0; i < vertexData.length; ++i) {
+        array[i] = vertexData[i];
+    }
+}
+
+const transformSize = 4 * 16;
+function populateTransformArray(array) {
+    let viewMatrix = mat4.create();
+    mat4.translate(viewMatrix, viewMatrix, vec3.fromValues(0, 0, -5));
+    let now = Date.now() / 1000;
+    mat4.rotate(viewMatrix, viewMatrix, 1, vec3.fromValues(Math.sin(now), Math.cos(now), 0));
+
+    let modelViewProjectionMatrix = mat4.create();
+    mat4.multiply(modelViewProjectionMatrix, projectionMatrix, viewMatrix);
+
+    for (let i = 0; i < 16; i++) {
+        array[i] = modelViewProjectionMatrix[i];
+    }
+}
+
+async function init() {
+    const adapter = await webgpu.requestAdapter({}); // Current (unifinished) implementation of requestAdapter requires this argument.
```

Comment:
`requestAdapter()` should probably work too, we should make the argument optional if it isn't already.